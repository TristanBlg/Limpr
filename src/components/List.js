import * as React from 'react'
import {Link} from 'react-router-dom'
import styled from 'styled-components'

function List(props){
  const {title, items} = props

  return (
    <StyledList>
      <h2 className="list-title">{title}</h2>
      <ul className="list-nav">
        {items && items.map(item => {
          return (
            <li key={item.id}>
              <Link className="list-link" to="/">
                {item.name}
              </Link>
              {item.author && (
                <React.Fragment>
                  &nbsp;-&nbsp;
                  <Link to="/" className="list-link list-link--high">
                    {item.author}
                  </Link>
                </React.Fragment>
              )}
            </li>
          )
        })}
      </ul>
    </StyledList>
  )
}

const StyledList = styled.div.attrs({
  className: 'list'
}) `
  margin-bottom: 40px;
  padding: 0 20px;

  .list{
    &-title{
      font-size: 1.6rem;
      font-weight: 700;
      color: #111;
      text-transform: uppercase;
      margin: 0;
    }
    &-nav{
      list-style: none;
      padding: 0;
      margin: 0;
    }
    &-link{
      font-size: 1.6rem;
      font-weight: 400;
      color: #111;

      &--high{
        color: #828282;
      }
    }
  }
`

export default List
