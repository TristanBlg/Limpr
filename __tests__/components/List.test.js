import React from 'react'
import {Link} from 'react-router-dom'
import {shallow} from '../../enzyme'

import List from '../../src/components/List'

describe('List tests', () => {
  it('renders title', () => {
    const title = "Title"
    const wrapper = shallow(<List title="Title" />)

    expect(wrapper.find('.list-title').text()).toEqual('Title')
  })

  it('renders list-items', () => {
    const items = [
      {
        "name": "Item 1",
        "author": "Author 1"
      },
      {
        "name": "Item 2",
        "author": "Author 2"
      },
      {
        "name": "Item 3",
        "author": "Author 3"
      }
    ]
    const wrapper = shallow(<List items={items} />)

    expect(wrapper.find('.list-nav').children()).toHaveLength(items.length)
  })

  it('renders a list item', () => {
    const items = [
      {
        "name": "Item 1"
      },
      {
        "name": "Item 2",
        "author": "Author 2"
      }
    ]
    const wrapper = shallow(<List items={items} />)

    expect(wrapper.contains(<li><Link to="/" className="list-link">Item 1</Link></li>)).toBeTruthy()
  })
})
