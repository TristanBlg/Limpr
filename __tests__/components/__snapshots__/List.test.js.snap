// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <List/> render the component 1`] = `
<div
  className="List__StyledList-cySCHs list gKxmxy"
>
  <h2
    className="list-title"
  >
    Title
  </h2>
  <ul
    className="list-nav"
  >
    <li>
      <a
        className="list-link"
        href="/"
        onClick={[Function]}
      >
        Item 1
      </a>
    </li>
    <li>
      <a
        className="list-link"
        href="/"
        onClick={[Function]}
      >
        Item 2
      </a>
    </li>
    <li>
      <a
        className="list-link"
        href="/"
        onClick={[Function]}
      >
        Item 3
      </a>
    </li>
  </ul>
</div>
`;
